plugins/syncbeacon/payload/payload.go

```diff
@@ -16,8 +17,15 @@ const (
 
 // Type is the type of the syncbeacon payload.
 var Type = payload.NewType(200, ObjectName, func(data []byte) (payload payload.Payload, err error) {
-	payload, _, err = FromBytes(data)
-
+	var consumedBytes int
+	payload, consumedBytes, err = FromBytes(data)
+	if err != nil {
+		return nil, err
+	}
+	if consumedBytes != len(data) {
+		return nil, xerrors.New("not all payload bytes were consumed")
+	}
+	return
```

Comment:
Is this a Github issue? I see two return

Reply:
my bad, pushed a fix